homeassistant/components/heos/media_player.py

```diff
@@ -235,10 +302,26 @@ async def async_update(self):
         controls = self._player.now_playing_media.supported_controls
         current_support = [CONTROL_TO_SUPPORT[control] for control in controls]
         self._supported_features = reduce(ior, current_support, BASE_SUPPORTED_FEATURES)
+        self._group = await self.async_get_group()
 
         if self._source_manager is None:
             self._source_manager = self.hass.data[HEOS_DOMAIN][DATA_SOURCE_MANAGER]
 
+    @log_command_error("unjoin_player")
+    async def async_unjoin_player(self):
+        """Remove this player from any group."""
+        if self.group_members:
```

Comment:
I would remove this check and simply call the API command. This will ensure the player is ungrouped should there have been an error previously populating the group members.

Reply:
You are right, removing that check makes it more robust.